components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -11650,6 +11650,16 @@ public static String retrieveDefaultReservedUsername() {
         return defaultReservedUsername;
     }
 
+    public static boolean isDefaultApplicationCreationEnabled() {
+        APIManagerConfiguration apiManagerConfiguration =
+                ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        String createDefaultApp = apiManagerConfiguration
+                .getFirstProperty(APIConstants.API_STORE_CREATE_DEFAULT_APPLICATION);
+        if (StringUtils.isNotEmpty(createDefaultApp)) {
+            return Boolean.parseBoolean(createDefaultApp);
+        }
+        return false;
```

Comment:
lets return true for default case. if config is not there we can set it to create default app

Reply:
Addressed via https://github.com/wso2/carbon-apimgt/pull/10498/commits/8616dc208127cc817d5ca029a43806768f6b650c